test/parallel/test-buffer-write-noassert.js

```diff
@@ -11,12 +11,21 @@ function write(funx, args, result, res) {
     assert.deepStrictEqual(buf, res);
   }
 
+  /*{
+    const invalidArgs = Array.from(args);
+    invalidArgs[1] = -1;
+    assert.throws(
+      () => Buffer.alloc(9)[funx](...invalidArgs),
+      /^RangeError: Index out of range$/
+    );
+  }*/
+
   {
     const invalidArgs = Array.from(args);
     invalidArgs[1] = -1;
     assert.throws(
       () => Buffer.alloc(9)[funx](...invalidArgs),
-      /^RangeError: (?:Index )?out of range(?: index)?$/
+      /^RangeError(?:\[ERR_INDEX_OUT_OF_RANGE\])?: Index out of range$/
```

Comment:
@jasnell If we *really* wanted to use `common.expectsError()` and not a `RegExp`, we could do something like the following (untested) code:

```js
let expectedError;
if (['writeFloatBE', 'writeFloatLE'].includes(funx)) {
  expectedError = common.expectsError(undefined, RangeError, 'Index out of range');
} else {
  expectedError = common.expectsError('ERR_INDEX_OUT_OF_RANGE', RangeError, 'Index out of range');
}
assert.throws(() => Buffer.alloc(9)[funx](...invalidArgs), expectedError);
```

Definitely more complicated, though, so I'm fine with the `RegExp` approach. Either way.